src/paint.rs

```diff
@@ -17,6 +21,40 @@ pub enum PaintFallback {
     ///
     /// [`<color>`]: https://www.w3.org/TR/css-color-3/
     Color(Color),
+    /// A `var()` function fallback.
+    /// 
+    /// See [`VariableFunction`] for details.
+    Variable(&'a str, Option<Box<PaintFallback<'a>>>),
```

Comment:
I'd like to draw attention to this, using a `Box` here prevents `PaintFallback` from being `Copy`. That's a breaking change. Is this ok or should I simply store `&str` when `var()` fallback also has a fallback?

Reply:
Started implementing in for `Angle` as well, adding this must be a breaking change.

It will also prevent a lot of other types (besides `Paint` and `Angle`) from being Copy because they will all need to be self, referential.

I'm not sure whether I should be eagerly parsing them though because browsers properly resolve rules like `var(--valid-color, hsl(some, input, error))`, because they use `--valid-color`. Parsing eagerly returns `Error`s if any of the fallbacks fail. So I'll probably add some `ParseAsNeeded` type that handles parsing for `T: ParseLater` when dereferenced or something along those lines.

Reply:
Scratch that, I see that most properties aren't parsed so, storing just `&str` is fine.